driver/src/main/java/org/neo4j/driver/internal/async/pool/ExtendedChannelPool.java

```diff
@@ -20,9 +20,13 @@
 
 import io.netty.channel.pool.ChannelPool;
 
+import java.util.concurrent.CompletionStage;
+
 public interface ExtendedChannelPool extends ChannelPool
```

Comment:
Wdyt about removing `extends ChannelPool` and exposing only the methods `ConnectionPoolImpl` really needs?
For example, this would hide `close()` and `closeAsync()` so `ConnectionPoolImpl` would only have a possibility to call `repeatableCloseAsync()`